src/openforms/registrations/contrib/camunda/plugin.py

```diff
@@ -149,14 +151,30 @@ def register_submission(
             }
         }
 
-    def get_reference_from_result(self, result: Dict[str, str]) -> NoReturn:
+    def get_reference_from_result(
+        self, result: Dict[str, str], submission: Submission
+    ) -> NoReturn:
         """
         Extract the public submission reference from the result data.
 
-        We never return, as the Camunda API response does not contain anything useful
-        and readable for the end-user and we cannot make assumptions about the process
-        model. The instance ID is a UUID, which is not suitable for end users.
+        Returning is deferred to third party plugins. If there is no suitable plugin
+        or plugin return value, we do not return at all because we cannot make any
+        assumptions. This results in Open Forms generating a reference.
         """
+        import bpdb
+
+        bpdb.set_trace()
+        options = self.get_options(submission.form)
+        if plugin_id := options.get("plugin"):
+            plugin = camunda_plugin_register[plugin_id]
+            client = get_client()
+            reference = plugin.obtain_reference(client, submission)
+            if reference:
+                logger.info("Plugin %r returned reference: %s", plugin, reference)
+                return reference
+            else:
+                logger.info("Plugin %r returned empty reference", plugin)
+
         raise NoSubmissionReference("Deferred to Open Forms itself")
```

Comment:
I think I'll move this up a level, as this does not entirely do semantically what I'd like it to. There should be:

1. registration result and references must be saved/persisted/committed after the result was received
2. then a post-registration hook should be invoked which can block the rest of the process before user-feedback is returned

Instead of applying side-effects in the `get_reference_from_result` - this really should just be concerned with data that has been persisted in the `result` dict already.